File: gis-account-service/src/main/java/si/um/feri/controllers/HealthController.java
package si.um.feri.controllers;

import io.smallrye.common.annotation.Blocking;
import org.jboss.resteasy.reactive.RestResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.core.MediaType;

@Path("/api/v1")
public class HealthController {

    private Logger logger = LoggerFactory.getLogger(HealthController.class);

    @GET
    @Path("/ping")
    @Blocking
    public Object pong() {
        logger.info("Connecting to Health Controller for Pong.");
        return RestResponse.ResponseBuilder.ok( "Pong", MediaType.TEXT_PLAIN_TYPE).build();
    }
}

File: gis-geodata-service/src/main/kotlin/si/um/feri/gisgeodataservice/controllers/HealthController.kt
package si.um.feri.gisgeodataservice.controllers

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatusCode
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1")
class HealthController {

    private var logger: Logger = LoggerFactory.getLogger(HealthController::class.java)

    @GetMapping("/ping")
    fun pong(): ResponseEntity<String>
    {
        logger.info("Connecting to Health Controller for Pong.")
        return ResponseEntity("Pong", HttpStatusCode.valueOf(200))
    }

}